### DISCUSSION TITLE
Consider an Animation Feature for Blocks

### DISCUSSION THREAD

**Discussion by beafialho** [WordPress/gutenberg#40160] [Ideas]
I know this idea has been discussed in the past, and it would be really cool to revisit an animation feature in the editor. The example in the video is from the [CoBlocks](https://wphelp.blog/how-to-add-animations-to-any-block-in-the-wordpress-editor/) plugin. Animations add a polished, subtle touch to websites and it would be great to have these work in any theme.

https://user-images.githubusercontent.com/65220155/162234582-d9d56922-1da4-484d-b385-b657f2b170a2.mp4

**Comment by aristath** [comment]
I'd be very hesitant to add animations like these to core blocks... Animations are fun when they are properly done, but in many cases they are an a11y issue, they have the potential to slow down a site on slower/older client machines and they can introduce a level of complexity that many visitors may find annoying. Animations are fancy, but in most cases don't really enhance the experience that much.
Of course we should differentiate "animations" from "micro-interations". A generic, grand animation that runs on page-load is a horrible thing to do. However, a subtle animation that runs on hover/click etc _can_ enhance user experience and that would be something worth considering.

**Comment by jasmussen** [comment]
One of the opportunities of having very basic animation features in core is to provide a mechanism that comes with basic accessibility affordances right out of the box, such as being turned off entirely in the case of reduce-motion flags detected, and by being non-interruptive in their playback (i.e. never blocking a user action until complete). In addition to this, it could provide points of extensibility on which plugins could build, extending those principles to plugins.

**Comment by aristath** [comment]
> One of the opportunities of having very basic animation features in core is to provide a mechanism that comes with basic accessibility affordances right out of the box, such as being turned off entirely in the case of reduce-motion flags detected, and by being non-interruptive in their playback (i.e. never blocking a user action until complete). In addition to this, it could provide points of extensibility on which plugins could build, extending those principles to plugins.
 
When framed like that, it makes sense... 👍

**Comment by davewhitley** [comment]
I'd like to add that if it's available in core, then we can guide the creator/designer in the right direction with motion options that are tasteful and not overdone, all implemented with respect to the viewers' preferences (which might not be the case with a plugin). If done well, motion can be a powerful tool for storytelling as well as enhanced usability via feedback and interaction.
